core/src/test/java/org/apache/accumulo/core/metadata/schema/TabletFileTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.metadata.schema;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.apache.accumulo.core.data.TableId;
+import org.junit.Test;
+
+public class TabletFileTest {
+
+  private void test(String metadataEntry, TableId tableId, String tabletDir, String fileName) {
+    TabletFile tabletFile = new TabletFile(metadataEntry);
+
+    assertEquals(metadataEntry, tabletFile.getMetadataEntry());
+    assertEquals(tableId, tabletFile.getTableId());
+    assertEquals(tabletDir, tabletFile.getTabletDir());
+    assertEquals(fileName, tabletFile.getFileName());
+
+  }
+
+  @Test
+  public void testValidPaths() {
+    test("hdfs://localhost:8020/accumulo/tables/2a/default_tablet/F0000070.rf", TableId.of("2a"),
+        "default_tablet", "F0000070.rf");
+    test("hdfs://nn1:9000/accumulo/tables/5a/t-0005/C0009.rf", TableId.of("5a"), "t-0005",
+        "C0009.rf");
+  }
+
+  @Test
+  public void testBadPaths() {
```

Comment:
I just tested these and noticed we need more validation on the volume.  Since we are traversing backwards up from the filename, anything before "tables" will be a valid volume.  Any ideas?  The best I can come up with is what I've seen... valid if ```path.contains(":")```

Reply:
Maybe ensure that `volumePath.toUri().getScheme()` is not null.

Reply:
That works great thanks!